src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -169,6 +175,11 @@ internal override bool IsDeclarationWithSharedBody(SyntaxNode declaration)
 
         protected override ImmutableArray<ISymbol> GetCapturedVariables(SemanticModel model, SyntaxNode memberBody)
         {
+            if (memberBody is CompilationUnitSyntax unit && unit.ContainsTopLevelStatements())
+            {
+                return model.AnalyzeDataFlow(((GlobalStatementSyntax)unit.Members[0]).Statement, ((GlobalStatementSyntax)unit.Members[^1]).Statement)!.Captured;
```

Comment:
Isn't it possible that the last member is a class declaration, then the cast to GlobalStatementSyntax fails?

e.g:

```
var x = ""; // cast is okay

public class C {} // will cast fail?
```

Reply:
🤦‍♂️

Thank you. I'm silly.

I was playing around with top level statements a few weeks ago, and tried the "Generate class" refactoring, which produced `private class C`, and was a compiler error, and so I thought you couldn't mix global statements and class declarations. Will definitely have to update this PR.

Reply:
> the "Generate class" refactoring, which produced `private class C`

Yeah the refactoring generates the class in the wrong place. This is tracked by https://github.com/dotnet/roslyn/issues/49649.